### PR TITLE
Update codespell and fix typos

### PR DESCRIPTION
**Pull request by KarshVashi**
The new version of codespell was failing with more detected typos that I fixed as this PR

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/www/test_utils.py [hunk 1]
```diff
@@ -266,7 +266,7 @@ def test_wrapped_markdown_with_raw_code_block(self):
         assert (
             """<div class="rich_doc" ><h1>Markdown code block</h1>
 <p>Inline <code>code</code> works well.</p>
-<pre><code>Code block\ndoes not\nrespect\nnewlines\n</code></pre>
+<pre><code>Code block\nodes not\nrespect\nnewlines\n</code></pre>
```
Comment (kaxil) [inline tests/www/test_utils.py]:
```suggestion
<pre><code>Code block\ndoes not\nrespect\nnewlines\n</code></pre>
```

File: .codespellignorelines [hunk 2]
```diff
@@ -1,3 +1,4 @@
             f"DELETE {source_table} FROM { ', '.join(_from_name(tbl) for tbl in stmt.froms) }"
         for frm in source_query.selectable.froms:
     roles = relationship("Role", secondary=assoc_user_role, backref="user", lazy="selectin")
+    The platform supports **C**reate, **R**ead, **U**pdate, and **D**elete operations on most resources.
```
Comment (kaxil) [inline .codespellignorelines:4]:
```suggestion
    The platform supports **C**reate, **R**ead, **U**pdate, and **D**elete operations on most resources.

<pre><code>Code block\ndoes not\nrespect\nnewlines\n</code></pre>
```

File: .codespellignorelines [hunk 3]
```diff
@@ -1,3 +1,6 @@
...
+
```
Comment (kaxil) [inline .codespellignorelines]:
```suggestion
```

**Review by kaxil — APPROVED** [review]

**Comment by pierrejeambrun** [comment]
Needs to be applied on top of https://github.com/apache/airflow/pull/27822 which will most certainly make it into 2.6.0 (improvement). Marking for 2.6.0 for now and skipping.

**Pull request merged**